src/services/authService.ts

```diff
@@ -23,29 +22,33 @@ export class AuthService implements IAuthService {
     async signInWithGoogle(): Promise<User | undefined> {
         try {
             const result = await signInWithPopup(this.auth, this.provider)
-            const userResponse = parseUser(result.user)
-            return await this.createUserOnFirstLogin(userResponse)
+            return this.retrieveUser(result.user)
         } catch (error) {
             console.error(error)
             this.alertHelper.alertError("Não foi possivel logar com o Google")
         }
     }
-    async createUserOnFirstLogin(userResponse: User): Promise<User | undefined> {
-        const findUser = await this.userService.findUser(userResponse)
 
-        if(!findUser){
-            this.userService.createUser(userResponse)
-            return(userResponse)
+    async retrieveUser(data: any): Promise<User | undefined> {
+        const existingUser = await this.userService.findUser(data.email)
+
+        if(!existingUser){
+            return this.createUserOnFirstLogin(data)
```

Comment:
Now we know it is the first login (because there is no existing user). So we can separate this part to the real `createUserOnFirstLogin`function